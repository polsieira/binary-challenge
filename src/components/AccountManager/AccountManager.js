import React, { Component } from 'react';
import Login from '../../containers/Login/Login';
import CreateAccount from '../../containers/CreateAccount/CreateAccount';
import './AccountManager.scss';
import { Route, NavLink } from 'react-router-dom';

class AccountManager extends Component {
  constructor() {
    super();
    this.state = {
      selected: 'login'
    }
  }

  changeSelected = (id) => {
    this.setState({ selected: id });
  }

  render() {
    return (
      <div className='AccountManager' >
        <div className='modal-container'>
          <ul className='modal-tabs'>
            <NavLink to='login' className={this.state.selected === 'login' ? 'form-navlink form-navlink--left' : 'form-navlink form-navlink--left unselected'} onClick={() => this.changeSelected('login')}>
              <li className='login_tab'> Log In</li>
            </NavLink>
            <NavLink to='/create-account' className={this.state.selected === 'create account' ? 'form-navlink form-navlink--right' : 'form-navlink form-navlink--right unselected'} onClick={() => this.changeSelected('create account')}>
              <li className='login_tab'>Create Account</li>
            </NavLink>
          </ul>
          <Route exact path='/login' render={() => <Login />}
          />
          <Route exact path='/create-account' render={() => <CreateAccount />} />
        </div>
      </div>
    );
  }
};

export default AccountManager;
