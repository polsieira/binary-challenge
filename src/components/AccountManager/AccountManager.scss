.AccountManager {
  background-image: url('https://images.unsplash.com/photo-1465101162946-4377e57745c3?ixlib=rb-1.2.1&auto=format&fit=crop&w=1657&q=80');
  background-repeat: no-repeat; 
  background-size: 100% 100%;
  color: white;
  display: flex;
  height: 100vh;
  justify-content: flex-end;
  width: 100%;

  .modal-container {
    background: white;
    height: 100vh;
    opacity: 0.95;
    position: absolute;
    width: 50%;

    .modal-tabs {
      display: flex;
      height: 13vh;
      justify-content: space-between;
      list-style: none;
      margin: 0px;
      padding: 0px;

      .form-navlink {
        color: black;
        display: flex;
        flex-direction: column;
        height: 100%;
        justify-content: center;
        text-decoration: none;
        width: 50%;
      }

      .form-navlink--right {
        border-left: 2px black solid;
      }
      
      .unselected {
        border-bottom: 2px black solid;
      }

    }
  }

  .background-image {
    background-size: 100% 100%;
  }
}