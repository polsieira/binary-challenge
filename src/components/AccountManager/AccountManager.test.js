import { AccountManager, mapDispatchToProps } from './AccountManager'
import { shallow } from 'enzyme';
import React from 'react';


describe('AccountManager', () => {

  let wrapper;

  describe('AccountManager component', () => {
    beforeEach(() => {
      wrapper = shallow(<AccountManager />)
    })

    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

});

