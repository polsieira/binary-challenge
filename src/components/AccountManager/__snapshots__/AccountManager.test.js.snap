// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountManager AccountManager component should match a snapshot 1`] = `
<div
  className="AccountManager"
>
  <div
    className="modal-container"
  >
    <ul
      className="modal-tabs"
    >
      <NavLink
        className="form-navlink form-navlink--left"
        onClick={[Function]}
        to="login"
      >
        <li
          className="login_tab"
        >
           Log In
        </li>
      </NavLink>
      <NavLink
        className="form-navlink form-navlink--right unselected"
        onClick={[Function]}
        to="/create-account"
      >
        <li
          className="login_tab"
        >
          Create Account
        </li>
      </NavLink>
    </ul>
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/create-account"
      render={[Function]}
    />
  </div>
</div>
`;
